vm-device/src/interrupt/mod.rs

```diff
@@ -141,6 +141,27 @@ pub trait InterruptSourceGroup: Send + Sync {
     #[allow(unused_variables)]
     fn notifier(&self, index: InterruptIndex) -> Option<EventFd>;
 
+    /// Enable the resamplefd.
+    /// With the resamplefd, vfio INTx interrupt can be automatically unamsk in EOI
+    #[cfg(target_arch = "x86_64")]
+    fn enable_resamplefd(&self, _index: InterruptIndex) -> Result<()> {
+        // Not all interrupt sources need resamplefd.
+        // If not, we can have a no-op here.
+        Ok(())
+    }
+
+    /// Returns an interrupt resample notifier from this interrupt.
+    ///
+    /// An interrupt resample notifier allows for kvm re-enable level triggered
+    /// interrupt automatically without userspace intervention.
+    #[allow(unused_variables)]
+    #[cfg(target_arch = "x86_64")]
+    fn unmask_notifier(&self, index: InterruptIndex) -> Option<EventFd> {
```

Comment:
Question (rather than a suggestion): why not name it `resamplefd_notifier()`? Do you see the need of other "notifiers" for unmask other than using `resamplefd`?

Reply:
This interface is only for resamplefd, I think resamplefd_notifier makes more sense.

Reply:
> This interface is only for resamplefd, I think resamplefd_notifier makes more sense.

@rbradford I have renamed the function name into resamplefd_notifier.